client.py

```diff
@@ -51,7 +51,7 @@ def parse_ServerResponse(self, response):
         assert x.tag == '{http://schemas.xmlsoap.org/soap/envelope/}Envelope'
         r = x.find('.//{http://ctkipservice.rsasecurity.com}' + outer)
         ad = r.find('{http://ctkipservice.rsasecurity.com}AuthData')
-        assert ad.text == self.auth == response.headers.get('Authorization')
+        assert ad.text == self.auth #== response.headers.get('Authorization')
```

Comment:
@dlenski I also encountered an assertion failure here.  The response was of the form:

```
<?xml version="1.0" encoding="UTF-8"?>
<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
  <soapenv:Body>
    <ServerResponse xmlns="http://ctkipservice.rsasecurity.com">
      <AuthData>XXX_ACTIVATION_CODE</AuthData>
      <ProvisioningData>XXX_Base64_of_another_xml_document_specifying_provisioning_version</ProvisioningData>
      <Response> XXX crapload of base64 here</Response>
    </ServerResponse>
  </soapenv:Body>
</soapenv:Envelope>
```

It looks like `ad.text` matches `self.auth` (i.e., supplied activation code).  Unfortunately I don't know what `response.headers.get('Authorization')` contained.

Reply:
Thanks, that's interesting. "My" server duplicated the `authdata` tag in the `Authorization` header, but I guess others don't.